<launch>
    <!--
    <param name="/use_sim_time" value="true"/>
    -->
    <arg name="scenario_num" default="1"/>
    
    
    <node pkg="pie" name="scenario_execution_uav0" type="scenario_execution.py" output="screen" clear_params="true">
        <param name="uav_num" value="0" />
        <param name="env_boundry_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/environment_boundary_config/" />
        <param name="scenario_num" value="$(arg scenario_num)" />
    </node>
<!--
    <node pkg="pie" name="true_scenario_uav0" type="true_scenario.py" output="screen" clear_params="true">
        <param name="uav_num" value="0" />
    </node>
-->
    <node pkg="pie" name="evaluator_uav0" type="evaluator.py" output="screen" clear_params="true">
        <param name="uav_num" value="0" />
        <param name="scaler_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/scaler.pkl" />
        <param name="encoder_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/encoder.pkl" />
        <param name="model_name" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/lstm_direction_2_layers_1_features_7_neurons_64_timesteps_128_epochs_290.h5"/>     
        <param name="time_step" value="128" />
    </node>
   
<!--
    <node pkg="pie" name="save_data_uav0" type="save_data.py" output="screen" clear_params="true">
        <param name="uav_num" value="0" />
        <param name="data_log_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/flightData/"/>
    </node>

 -->
    
    
    

    <node pkg="pie" name="scenario_execution_uav1" type="scenario_execution.py" output="screen" clear_params="true">
        <param name="uav_num" value="1" />
        <param name="env_boundry_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/environment_boundary_config/" />
        <param name="scenario_num" value="$(arg scenario_num)" />
    </node>
<!--
    <node pkg="pie" name="true_scenario_uav1" type="true_scenario.py" output="screen" clear_params="true">
        <param name="uav_num" value="1" />
    </node>
-->
    <node pkg="pie" name="evaluator_uav1" type="evaluator.py" output="screen" clear_params="true">
        <param name="uav_num" value="1" />
        <param name="scaler_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/scaler.pkl" />
        <param name="encoder_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/encoder.pkl" />
        <param name="model_name" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/lstm_direction_2_layers_1_features_7_neurons_64_timesteps_128_epochs_290.h5"/>     
        <param name="time_step" value="128" />
    </node>

    


    <node pkg="pie" name="scenario_execution_uav2" type="scenario_execution.py" output="screen" clear_params="true">
        <param name="uav_num" value="2" />
        <param name="env_boundry_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/environment_boundary_config/" />
        <param name="scenario_num" value="$(arg scenario_num)" />
    </node>
<!--
    <node pkg="pie" name="true_scenario_uav2" type="true_scenario.py" output="screen" clear_params="true">
        <param name="uav_num" value="2" />
    </node>
-->
    <node pkg="pie" name="evaluator_uav2" type="evaluator.py" output="screen" clear_params="true">
        <param name="uav_num" value="2" />
        <param name="scaler_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/scaler.pkl" />
        <param name="encoder_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/encoder.pkl" />
        <param name="model_name" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/lstm_direction_2_layers_1_features_7_neurons_64_timesteps_128_epochs_290.h5"/>     
        <param name="time_step" value="128" />
    </node>

    

    <node pkg="pie" name="scenario_execution_uav3" type="scenario_execution.py" output="screen" clear_params="true">
        <param name="uav_num" value="3" />
        <param name="env_boundry_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/environment_boundary_config/" />
        <param name="scenario_num" value="$(arg scenario_num)" />
    </node>
<!--
    <node pkg="pie" name="true_scenario_uav3" type="true_scenario.py" output="screen" clear_params="true">
        <param name="uav_num" value="3" />
    </node>
-->
    <node pkg="pie" name="evaluator_uav3" type="evaluator.py" output="screen" clear_params="true">
        <param name="uav_num" value="3" />
        <param name="scaler_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/scaler.pkl" />
        <param name="encoder_path" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/encoder.pkl" />
        <param name="model_name" value="/media/ariac/DATAPART1/mrinmoys-document/test-and-eval-of-multi-agent-system/simulation_ws/src/pie/script/models/lstm_direction_2_layers_1_features_7_neurons_64_timesteps_128_epochs_290.h5"/>     
        <param name="time_step" value="128" />
    </node>

    

</launch>